faunadb-java/src/test/java/com/faunadb/client/ClientSpec.java

```diff
@@ -2086,6 +2086,33 @@ public void shouldTestIdentity() throws Exception {
     }
   }
 
+  @Test
+  public void shouldTestCreateAccessProvider() throws Exception {
+    String roleName = randomStartingWith("role_");
+    String providerName = randomStartingWith("provider_");
+    String issuerName = randomStartingWith("issuer_");
+    String fullUri = randomStartingWith("https://") + ".auth0.com";
+
+    adminClient.query(CreateRole(Obj(
+      "name", Value(roleName),
+      "privileges", Obj(
+        "resource", Databases(),
+        "actions", Obj("read", Value(true))
+      )
+    ))).get();
+
+    Value provider = adminClient.query(
+      CreateAccessProvider(Obj(
+      "name", Value(providerName),
+      "issuer" , Value(issuerName),
+      "jwks_uri" , Value(fullUri),
+      "roles", Arr(Role(roleName))))
+    ).get();
+
+    List<RefV> roles = provider.at("roles").collect(Field.as(REF));
+    assertThat(roles.size(), equalTo(1));
```

Comment:
Could we check for the rest of the fields here as well (e.g, `name`, `issuer`, etc)? :)

There's also a _dynamic_ field called `audience` which is added to the `AccessProvider` upon creation. Since its value would be a bit more tricky to determine before creation, we could just check if the value is present in the response. 

Something along the lines:

```java
assertThat(provider.at("audience").getOptional(), is(Optional.<Value>empty()));
```